storage/ndb/src/kernel/vm/mt.cpp

```diff
@@ -8067,76 +8054,46 @@ mt_add_thr_map(Uint32 block, Uint32 instance)
   case DBTUX:
   case BACKUP:
   case RESTORE:
-    if (receive_threads_only)
-    {
-      thr_no += (instance - 1);
-    }
-    else
-    {
-      thr_no += (instance - 1);
-    }
+    thr_no += (instance - 1);
     break;
   case DBQLQH:
   case DBQACC:
   case DBQTUP:
   case DBQTUX:
   case QBACKUP:
   case QRESTORE:
-    if (receive_threads_only)
-    {
-      thr_no += (instance - 1);
-    }
-    else
-    {
-      thr_no += (instance - 1);
-    }
+    thr_no += (instance - 1);
     break;
   case PGMAN:
-    if (instance == num_lqh_threads + 1)
+    if (instance == num_lqh_workers + 1)
     {
       // Put extra PGMAN together with it's Proxy
       thr_no = block2ThreadId(block, 0);
     }
     else
     {
-      thr_no += (instance - 1) % num_lqh_threads;
+      thr_no += (instance - 1);
```

Comment:
These blocks are handled by ldm threads previously. Should we handle the situation that (ldm threads num == 0 && tc threads num > 0) here?

Reply:
isn't allowed to happen that num_ldm_threads == 0 AND num_tc_threads > 0